inspire/base/static/js/search/search_results.js

```diff
@@ -44,8 +44,188 @@ require(['jquery', 'bootstrap'], function($) {
+    });
+  });
+
+  var sList = [];
+  $('#checkbox-select-all').click(function(event) {
+    if (this.checked) {
+      $('.checkbox-results').each(function() {
+        this.checked = true;
+        sList.push($(this).attr("id"));
+      });
+
+      if (parseInt($('#total-results').text()) > 5000) {
+        $('#results-panel').after('<div class="panel panel-default" id="info-message">' +
+          '<div class="panel-body" >You have selected ' + filter_list(sList).length + ' records of this page. <a class="pointer" id="select-all-records">' +
+          'Select 5000 records (Maximum limit).</a></div></div>');
+      } else {
+        $('#results-panel').after('<div class="panel panel-default" id="info-message">' +
+          '<div class="panel-body" >You have selected ' + filter_list(sList).length + ' records of this page. <a class="pointer" id="select-all-records">' +
+          'Select all ' + $('#total-results').text() + ' results.</a></div></div>');
+      }
+
+      $('#select-all-records').on('click', function() {
+        $('#info-message').remove();
+        $.get("/search?of=id&rg=" + $('#total-results').text(), function(data, status) {
+          if (status == 'success') {
+            if (data.length > 5000) {
+              data.splice(5000, data.length);
+              $('#results-panel').after('<div class="alert alert-warning" id="alert-selection" role="alert">' +
+                '5000 records have been selected.<a class="pointer" id="undo-selection"> Undo selection.</a></div>');
+            } else {
+              $('#results-panel').after('<div class="alert alert-warning" id="alert-selection" role="alert">' +
+                $('#total-results').text() + ' records have been selected.<a class="pointer" id="undo-selection"> Undo selection.</a></div>');
+            }
+            sList = data;
+          }
+        });
+      });
+    } else {
+      $('.checkbox-results').each(function() {
+        this.checked = false;
+      });
+      sList = [];
+      $('#info-message').remove();
+      $('#alert-selection').remove();
+    }
+  });
+
+  $(document).on('click', '#undo-selection', function(e) {
+    sList = [];
+    $('#alert-selection').remove();
+    $('.checkbox-results').each(function() {
+      this.checked = false;
+    });
+    $('#checkbox-select-all').prop('checked', false);
+  });
+
+
+  $('#checkbox-parent > input[type=checkbox]').change(function() {
+    var $this = $(this);
+    if (this.checked) {
+      sList.push($this.attr("id"));
+    } else {
+      var index = sList.indexOf($this.attr("id"))
+      if (index > -1) {
+        sList.splice(index, 1);
+      }
+    }
+    if (sList != 0) {
+      $("#dropdown-export").html($('#dropdown-export').text().trim() + ' <span class="caret"></span>');
+    }
+  });
+
+  var $loading = $('#spinner-download').hide();
+  $("#download-format").on('click', function() {
+    url = create_url($('#dropdown-export').text().trim());
```

Comment:
Combine `create_url` and `get_format` into one function that returns an object with both.